### DISCUSSION TITLE
Realtime event coming multiple times

### DISCUSSION THREAD

**Discussion by shellking4** [pocketbase/pocketbase#2736] [Q&A]
How can I avoid the realtime events being sent multiple times ? I noticed that sometimes, in quite a random fashion, the realtime event would be sent multiple times to my Svelte app. Since I'm doing a price update based on the event, data is biased. How can I implement some kind of idempotence for this ?

**Comment by ganigeorgiev** [comment]
>  I noticed that sometimes, in quite a random fashion, the realtime event would be sent multiple times to my Svelte app. 

I don't know how to reproduce it and I don't think it is possible, because we don't store the events on the server.

The only way to receive a single event multiple times is if you have subscribed multiple times to the same topic.

Could you provide a code sample or ideally a minimal reproducible repo?

Also keep in mind that real time events are sent for every record change. For example, if you are updating a single record by performing multiple requests, you'll receive an event for each successful request.

**Reply by shellking4** [reply]
Thank you for your reply @ganigeorgiev 
I console log in the subscribe handler again and I can see it's being sent once. Maybe I'm doing something wrong in my code...
